test-data/unit/check-statements.test

```diff
@@ -2195,6 +2195,14 @@ class B: pass
 def foo(x: int) -> Union[Generator[A, None, None], Generator[B, None, None]]:
     yield x  # E: Incompatible types in "yield" (actual type "int", expected type "Union[A, B]")
 
+[case testReturnTypeOfGeneratorUnion]
+from typing import Generator, Union
+
+class T: pass
+
+def foo(arg: Union[Generator[int, None, T], Generator[str, None, T]]) -> Generator[Union[int, str], None, T]:
```

Comment:
What about a union of non-generator types?

Reply:
Yield from a union of non-generator types returns an error (yield from can't be applied to) on line 5100, I added a test for that now.